workers/unity/Assets/Playground/Scripts/Cubes/CollisionProcessSystem.cs

```diff
@@ -0,0 +1,76 @@
+using Generated.Playground;
+using Improbable.Gdk.Core;
+using Unity.Collections;
+using Unity.Entities;
+
+namespace Playground
+{
+    [RemoveAtEndOfTick]
+    public struct CollisionComponent : IComponentData
+    {
+        public Entity ownEntity;
+        public Entity otherEntity;
+    }
+
+    [UpdateInGroup(typeof(SpatialOSUpdateGroup))]
+    internal class CollisionProcessSystem : ComponentSystem
+    {
+        private struct Data
+        {
+            public readonly int Length;
+            public ComponentDataArray<CollisionComponent> Collision;
+            public EntityArray Entities;
+            public ComponentDataArray<SpatialOSLaunchable> Launchable;
+            [ReadOnly] public ComponentDataArray<CommandRequestSender<SpatialOSLauncher>> Sender;
+            [ReadOnly] public ComponentDataArray<Authoritative<SpatialOSLaunchable>> DenotesAuthority;
+        }
+
+        [Inject] private Data data;
+
+        protected override void OnUpdate()
+        {
+            for (var i = 0; i < data.Length; i++)
+            {
+                // Handle all the different possible outcomes of the collision.
+                // This requires looking at their most recent launchers.
+                var collision = data.Collision[i];
+                var first = data.Collision[i].ownEntity;
+                var second = data.Collision[i].otherEntity;
+                var first_launchable = EntityManager.GetComponentData<SpatialOSLaunchable>(first);
+                var second_launchable = EntityManager.GetComponentData<SpatialOSLaunchable>(second);
+                var first_launcher = first_launchable.MostRecentLauncher;
+                var second_launcher = second_launchable.MostRecentLauncher;
+                if (first_launcher == second_launcher)
+                {
+                    if (first_launcher != 0)
+                    {
+                        data.Sender[i].SendIncreaseScoreRequest(first_launcher,
+                            new Generated.Playground.ScoreIncreaseRequest()
```

Comment:
do we need the whole namespace or can we just write `new ScoreIncreaseRequest()`

Reply:
We need it. Not sure why.

Reply:
found out why: if you just use `new ScoreIncreaseRequest()`, it will pick the wrong code-generated struct, not our wrapper. in that case it's fine